[Merged by Bors] - Feat: port algebra.regular.smul

arienmalec opened the pull request:
Port of algebra.regular.smul

Based on 550b58538991c8977703fdeb7c9d51a5aa27df11

kbuzzard commented:
Should this file be called `Smul.lean` or `SMul.lean`? We have two contrasting conventions in this PR.

arienmalec commented:
> Should this file be called Smul.lean or SMul.lean? We have two contrasting conventions in this PR.

Should be `SMul.lean`; I created it wrongly, changed it locally, but the rename didn't take, so I changed it on Github and hopefully it took?

dupuisf commented on Mathlib/Algebra/Regular/SMul.lean:
```diff
@@ -0,0 +1,261 @@
+`M → M`, defined by `m ↦ a • m`, is injective.
+
+This property is the direct generalization to modules of the property `IsLeftRegular` defined in
+`Algebra/Regular`.  Lemma `IsSMulRegular.is_left_regular_iff` shows that indeed the two notions
+coincide.
```
```suggestion
`Algebra/Regular`.  Lemma `isLeftRegular_iff` shows that indeed the two notions
coincide.
```

dupuisf commented on Mathlib/Algebra/Regular/SMul.lean:
```diff
@@ -0,0 +1,261 @@
+
+variable (M)
+
+/-- One is `M`-regular always. -/
```
```suggestion
/-- One is always `M`-regular. -/
```

dupuisf commented on Mathlib/Algebra/Regular/SMul.lean:
```diff
@@ -0,0 +1,261 @@
+variable {G : Type _} [Group G]
+
+/-- An element of a group acting on a Type is regular. This relies on the availability
+of the inverse given by groups, since there is no `left_cancel_smul` typeclass. -/
```
```suggestion
of the inverse given by groups, since there is no `LeftCancelSMul` typeclass. -/
```

dupuisf reviewed (commented):
Just a few docstring changes, otherwise LGTM!

bors d+

arienmalec commented:
bors r+
